### PR TITLE
CODENVY-1407: Log agent output on fail

### PR DESCRIPTION
**Pull request by mmorhun**
### What does this PR do?
Logs agent output if it fails to start.

### What issues does this PR fix or reference?
https://github.com/codenvy/codenvy/issues/1407

#### Changelog
Add output of failed agent into tomcat logs.

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

File: agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java [hunk 1]
```diff
@@ -66,13 +68,33 @@ public AbstractAgentLauncher(long agentMaxStartTimeMs,
...
+     * @throws ServerException
+     *         if script execution failed
+     */
     @Override
     public void launch(Instance machine, Agent agent) throws ServerException {
         if (isNullOrEmpty(agent.getScript())) {
             return;
         }
+        ListLineConsumer agentLogger = new ListLineConsumer();
+        LineConsumer lineConsumer = new AbstractLineConsumer() {
```
Comment (garagatyi) [inline agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java:77]: CompositeLineConsumer?

**Comment by garagatyi** [comment]
Can you show an example of output if ssh connection fails?

File: agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java [hunk 2]
```diff
@@ -66,13 +68,33 @@ public AbstractAgentLauncher(long agentMaxStartTimeMs,
...
+     * The machine should be started.
+     * Retrieves agent's logs if its start fails.
+     *
+     * @param machine
+     *         the machine instance in which agent will be started. The machine should be started.
+     * @param agent
+     *         the agent to start
+     * @throws ServerException
+     *         if script execution failed
+     */
```
Comment (tolusha) [inline agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java]: Doc should be moved to interface.

File: agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java (hunk 1 above)
Comment (mmorhun) [reply agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java:77]: Possible, but in that case we shouldn't close `CompositeLineConsumer`, only `agentLogger` (which is part of composite consumer).

File: agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java [hunk 3]
```diff
@@ -66,13 +68,33 @@ public AbstractAgentLauncher(long agentMaxStartTimeMs,
         this.agentLaunchingChecker = agentLaunchingChecker;
     }
 
+    /**
+     * Launches agent script in machine and waits until agent starts.
+     * The machine should be started.
+     * Retrieves agent's logs if its start fails.
```
Comment (garagatyi) [inline agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java]: As far as I can see it logs agent logs if it fails, but stores it in any case. Is it needed to add logging related logic into javadocs?

**Review by garagatyi — APPROVED** [review]
I'm not big fan of putting agents logs into memory and into logs, but ok with that if we need it.
@tolusha Please take a look

**Review by tolusha — APPROVED** [review]

**Pull request merged**